packages/xstate-wallet/src/store/memory-channel-storage.ts

```diff
@@ -68,6 +68,18 @@ export class MemoryChannelStoreEntry implements ChannelStoreEntry {
     return !!this._supported;
   }
 
+  get isFinalized() {
+    return this.isSupported && this.supported.isFinal;
+  }
+
+  get finalizationProof(): SignedState[] {
+    if (!this.isFinalized) {
+      throw new Error('Channel is not finalized');
+    }
+    return this.sortedByDescendingTurnNum
+      .filter(s => s.isFinal)
+      .map(s => ({...this.channelConstants, ...s}));
```

Comment:
Shouldn't this just return `this.supported` when `this.isFinalized` is true?

Reply:
I think that only works when everyone signs a single final state. If we're dealing with the case where everyone is signing their own state I think we need to provide both states in the finalization proof.

Reply:
Right.

There's no guarantee that these states form a support proof though:
- `this.supported` might be supported by signatures on _it_
- It could also depend on signatures on another state (probably just the previous one)

There should be a `this.support` method which returns the support.


Also, I would recommend that the `conclude-channel` protocol has everyone sign the same final state.
- It's simple enough to reason about
- It means we can use `supportState`
- it's more gas-efficient: less call data, no `validTransition` calls, and no `whoSignedWhat` logic

Reply:
Yeah, I should have tagged this with a `TODO`. This isn't a proper check but relies on participants playing nicely.